contracts/governance/Staking/Staking.sol

```diff
@@ -297,22 +297,126 @@ contract Staking is
         _withdrawNext(until, receiver, true);
     }
 
+    /**
+     * @notice Withdraw the given amount of tokens.
+     * @param amount The number of tokens to withdraw.
+     * @param until The date until which the tokens were staked.
+     * @param receiver The receiver of the tokens. If not specified, send to the msg.sender
+     * @param vesting The vesting contract address.
+     * @dev Can be invoked only by whitelisted contract passed to governanceWithdrawVesting
+     * */
+    function _governanceWithdrawDirect(
+        uint96 amount,
+        uint256 until,
+        address receiver,
+        address vesting
+    ) internal {
+        _notSameBlockAsStakingCheckpoint(until);
+
+        _withdrawDirect(amount, until, receiver, vesting);
+        // @dev withdraws tokens for lock date 2 weeks later than given lock date if sender is a contract
+        //		we don't need to check block.timestamp here
+        _withdrawNext(until, receiver, true);
```

Comment:
```suggestion
        _withdrawNextDirect(until, receiver, true);
```

Reply:
Solved in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/447/commits/0a232d703e2b2581ed28ba276a58fe252c1ce872)